lib/impure/nre.nim

```diff
@@ -23,46 +23,38 @@ when defined(js):
 ## **Note**: If you love `sequtils.toSeq` we have bad news for you. This
 ## library doesn't work with it due to documented compiler limitations. As
 ## a workaround, use this:
-##
-## .. code-block:: nim
-##
-##    import std/nre except toSeq
-##
-##
+runnableExamples:
+  # either `import std/nre except toSeq` or fully qualify `sequtils.toSeq`:
+  import std/sequtils
+  iterator iota(n: int): int =
+    for i in 0..<n: yield i
+  assert sequtils.toSeq(iota(3)) == @[0, 1, 2]
 ## Licencing
 ## ---------
 ##
 ## PCRE has `some additional terms`_ that you must agree to in order to use
 ## this module.
 ##
 ## .. _`some additional terms`: http://pcre.sourceforge.net/license.txt
-##
 runnableExamples:
+  import std/sugar
   let vowels = re"[aeoui]"
-
-  let expectedResults = [
-    1 .. 1,
-    2 .. 2,
-    4 .. 4,
-    6 .. 6,
-    7 .. 7,
-  ]
-  var i = 0
-  for match in "moigagoo".findIter(vowels):
-    doAssert match.matchBounds == expectedResults[i]
-    inc i
+  assert collect(for match in "moiga".findIter(vowels): match.matchBounds) ==
+    @[1 .. 1, 2 .. 2, 4 .. 4]
+  from std/sequtils import toSeq
+  let s = sequtils.toSeq("moiga".findIter(vowels)) # fully qualified to avoid confusion with nre.toSeq
```

Comment:
What are the benefits of this change?

Reply:
collect makes tests shorter and more readable

Reply:
How so? It certainly makes the code shorter, but now the example requires the reader to know what `collect` does. 

Additionally, placing multiple complex elements all on one line, along with a surrounding function call, is generally considered less readable than spreading them out over multiple lines.

Reply:
`collect` is already used in examples for other modules, eg os, sugar, sequtils, lists, for good reasons. 

> Additionally, placing multiple complex elements all on one line

fixed